jep/0000/README.adoc

```diff
@@ -0,0 +1,246 @@
+proposes switching everything in Jenkins core to UTF-8.
+The proposed change is far from complete, however;
+notably, it does not address encoding of build step output.
+
+This JEP does not attempt to solve those issues,
+but neither does it preclude or impede their solution.
+The bulk of the work here involves the `durable-task` plugin
+which is not used by freestyle builds.
+
+=== Transcoding of process output from `Launcher`
+
+Non-durable task output from external processes
+(using the core `Launcher` interface, such as `checkout` using CLI Git)
+is not transcoded.
+If the process produces non-ASCII, non-UTF-8 output,
+and the process output is streamed to the build log,
+this could result in mojibake in the log.
+
+This likely needs to be addressed in Jenkins core
+by having `ProcStarter.stdout(Listener)` transcode as needed.
+In practice only a small amount of text sent to Pipeline build logs is produced this way,
+and it is particularly unlikely to be produced by user-controlled inputs
+and thus to contain any non-ASCII characters.
+
+Due to the low priority, this JEP does not currently attempt to fix this core issue.
+Addressing it _may_ help apply UTF-8 to freestyle build logs in the future.
+
+=== Default value for `DurableTaskStep.encoding`
+
+The historical default for `encoding` parameter to `sh` and similar steps was UTF-8.
+This did not match the default for `readFile` and `writeFile`,
+and anyway did not apply to output streamed to the build log
+(only to the `returnStdout` option).
+
+Retaining this default would improve compatibility in only limited circumstances,
+at the expense of doing the wrong thing in typical situations
+where a process is printing text in the system default encoding.
+
+=== Use of agent system default encoding
+
+As noted in the historical section,
+it is impossible for Pipeline builds to follow the same policy as freestyle builds
+of using the same encoding as the agent system default for the build log:
+there may be multiple agents in use, or none,
+and the particular agent(s) used are not even known when the build starts.
+
+=== Use of historical non-Unicode encodings
+
+Some users speaking non-English languages, particularly with non-Latin scripts,
+may prefer to work exclusively with text documents in a traditional encoding.
+This seems to be particularly commonplace in Japan.
+@kohsuke (pers. comm.) has mentioned this concern in the past.
+
+Since build logs were already rendered in UTF-8 via HTTP,
+the internal coding system should really only matter
+to users accessing the `$JENKINS_HOME/jobs/…/builds/…/log` file directly.
+While that may have been common in the early days of Hudson,
+this is likely a rare use case today.
+
+In any event, making Jenkins architecture more complex and potentially buggy
+to satisfy a marginal requirement seems a poor trade-off.
+
+== Backwards Compatibility
+
+Jenkins systems running on UTF-8-based computers
```

Comment:
And Jenkins systems running without UTF-8?  The *very precise* wording here makes me nervous because it's emphasizing that it'll work in one particularly common case but is not making other promises beyond that.

Reply:
I am just trying to start the compatibility section with the broad point that if you set UTF-8 as your system encoding everywhere, you can ignore the subsections. Will try to clarify.